### ISSUE TITLE
Add autocompleting textbox / dropdowns

### ISSUE THREAD

**Issue by signebedi** [signebedi/libreForms#62]
Eg. https://stackoverflow.com/questions/28308574/how-to-style-autocomplete-dropdown-list

**Comment by signebedi** [comment]
```
	<style>
		input[type=search] {
			background: none;
			font-weight: bold;
			border-color: #2e2e2e;
			border-style: solid;
			border-width: 2px 2px 2px 2px;
			outline: none;
			padding: 10px 20px 10px 20px;
			width: 250px;
		}
			/*Css to target the dropdownbox*/
		ul.ui-autocomplete {
			color: red !important;
			-moz-border-radius: 15px;
			border-radius: 15px;
		}
	</style>
	<link rel="stylesheet" href="//code.jquery.com/ui/1.11.2/themes/smoothness/jquery-ui.css">
	<script src="//code.jquery.com/jquery-1.10.2.js"></script>
	<script src="//code.jquery.com/ui/1.11.2/jquery-ui.js"></script>
	<link rel="stylesheet" href="/resources/demos/style.css">
	<script>
		$(function () {
			$("#search_input").autocomplete({
				source: {{ list_of_cities | safe }}
			});
		});
	</script>

```

```
	<form id="form1">
		<div class="ui-widget">
			<input name="city" id="search_input" type="search" placeholder="City, ST">
		</div>
	</form>
```

**Comment by signebedi** [comment]
See also:

1. https://jsfiddle.net/bootstrapious/6xac37nm
2. https://www.w3schools.com/howto/howto_js_autocomplete.asp

**Comment by signebedi** [comment]
Okay, this is probably a better, working homunculus:

```
<style>

.autocomplete-items {
  position: absolute;
  border-bottom: none;
  border-top: none;
  z-index: 99;
  top: 100%;
  left: 0;
  right: 0;
}

</style>

<!--Make sure the form has the autocomplete function switched off:-->
<form autocomplete="off">
    <div class="form-group">
        <input id="myInput" class="form-label mt-4 autocomplete default" type="text" name="myCountry" placeholder="Country">
    </div>
</form>
  
  <script>
  function autocomplete(inp, arr) {
    /*the autocomplete function takes two arguments,
    the text field element and an array of possible autocompleted values:*/
    var currentFocus;
    /*execute a function when someone writes in the text field:*/
    inp.addEventListener("input", function(e) {
        var a, b, i, val = this.value;
        /*close any already open lists of autocompleted values*/
        closeAllLists();
        if (!val) { return false;}
        currentFocus = -1;
        /*create a DIV element that will contain the items (values):*/
        a = document.createElement("DIV");
        a.setAttribute("id", this.id + "autocomplete-list");
        a.setAttribute("class", "autocomplete-items");
        /*append the DIV element as a child of the autocomplete container:*/
        this.parentNode.appendChild(a);
        /*for each item in the array...*/
        for (i = 0; i < arr.length; i++) {
          /*check if the item starts with the same letters as the text field value:*/
          if (arr[i].substr(0, val.length).toUpperCase() == val.toUpperCase()) {
            /*create a DIV element for each matching element:*/
            b = document.createElement("DIV");
            /*make the matching letters bold:*/
            b.innerHTML = "<strong>" + arr[i].substr(0, val.length) + "</strong>";
            b.innerHTML += arr[i].substr(val.length);
            /*insert a input field that will hold the current array item's value:*/
            b.innerHTML += "<input type='hidden' value='" + arr[i] + "'>";
            /*execute a function when someone clicks on the item value (DIV element):*/
            b.addEventListener("click", function(e) {
                /*insert the value for the autocomplete text field:*/
                inp.value = this.getElementsByTagName("input")[0].value;
                /*close the list of autocompleted values,
                (or any other open lists of autocompleted values:*/
                closeAllLists();
            });
            a.appendChild(b);
          }
        }
    });
    /*execute a function presses a key on the keyboard:*/
    inp.addEventListener("keydown", function(e) {
        var x = document.getElementById(this.id + "autocomplete-list");
        if (x) x = x.getElementsByTagName("div");
        if (e.keyCode == 40) {
          /*If the arrow DOWN key is pressed,
          increase the currentFocus variable:*/
          currentFocus++;
          /*and and make the current item more visible:*/
          addActive(x);
        } else if (e.keyCode == 38) { //up
          /*If the arrow UP key is pressed,
          decrease the currentFocus variable:*/
          currentFocus--;
          /*and and make the current item more visible:*/
          addActive(x);
        } else if (e.keyCode == 13) {
          /*If the ENTER key is pressed, prevent the form from being submitted,*/
          e.preventDefault();
          if (currentFocus > -1) {
            /*and simulate a click on the "active" item:*/
            if (x) x[currentFocus].click();
          }
        }
    });
    function addActive(x) {
      /*a function to classify an item as "active":*/
      if (!x) return false;
      /*start by removing the "active" class on all items:*/
      removeActive(x);
      if (currentFocus >= x.length) currentFocus = 0;
      if (currentFocus < 0) currentFocus = (x.length - 1);
      /*add class "autocomplete-active":*/
      x[currentFocus].classList.add("autocomplete-active");
    }
    function removeActive(x) {
      /*a function to remove the "active" class from all autocomplete items:*/
      for (var i = 0; i < x.length; i++) {
        x[i].classList.remove("autocomplete-active");
      }
    }
    function closeAllLists(elmnt) {
      /*close all autocomplete lists in the document,
      except the one passed as an argument:*/
      var x = document.getElementsByClassName("autocomplete-items");
      for (var i = 0; i < x.length; i++) {
        if (elmnt != x[i] && elmnt != inp) {
          x[i].parentNode.removeChild(x[i]);
        }
      }
    }
    /*execute a function when someone clicks in the document:*/
    document.addEventListener("click", function (e) {
        closeAllLists(e.target);
    });
  }
  
  /*An array containing all the country names in the world:*/
  var countries = {{ list_of_options | safe }};
  /*initiate the autocomplete function on the "myInput" element, and pass along the countries array as possible autocomplete values:*/
  autocomplete(document.getElementById("myInput"), countries);
  </script>

```

**Comment by signebedi** [comment]
Preliminarily implemented in a676c284b802fca3cfc548ea85a31a221906aeb1. However, we still need to fix the appearance of the dropdown menus.

**Comment by signebedi** [comment]
Addressed issues with the appearance / location of the autocomplete dropdown in b3e34d7c6bab6d6dc27e6f6b3bc8d89ffc12d1ea.
